engine/jobmaster/dm/metadata/job.go

```diff
@@ -127,3 +127,30 @@ func (jobStore *JobStore) UpdateStages(ctx context.Context, taskIDs []string, st
 
 	return jobStore.Put(ctx, job)
 }
+
+// UpdateConfig will be called if user update job config.
+func (jobStore *JobStore) UpdateConfig(ctx context.Context, jobCfg *config.JobCfg) error {
+	state, err := jobStore.Get(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	oldJob := state.(*Job)
+
+	// TODO: each task can have different version.
```

Comment:
the purpose of this TODO is to reduce worker shutdown and restart? need to leverage pros and cons

Reply:
Yes,  changing the configuration of one task will not affect other tasks. Add more comment in [9a868af](https://github.com/pingcap/tiflow/pull/6508/commits/9a868afdd2d787e84b51c10165f535b5132af4d4)